### PR TITLE
Fix errors F.det and F.batch_det raises

### LINKED ISSUE #986: F.det should not raise error in forward propagation when input matrix is singular
Currently, `F.det` raises `ValueError` in forward propagation when input matrix is singular [here](https://github.com/pfnet/chainer/pull/613/files#diff-4d0027dd8b2ecc0d9eae073ddfefdb26R65).

It is beneficial when we do backward propagation because it calculates the inverse of the input matrix. But some users can use `F.det` without backward propagation, in which case singular matrices are not exceptional. So we should not raise errors.

### PR DESCRIPTION
**Pull request by delta2323**
* Add `info` vector to the return value of `_inv_gpu`.
* Change `success` with `info` in the return value of `_det_gpu`
* `F.det` and `F.batch_det` does not raise `ValueError` even if some input matrices are singular.
* `F.det` and `F.batch_det` raise `numpy.LinAlgError`when input matrices are singular.

Fix #986 

Please merge #988 before this PR.

### CODE REVIEW

File: chainer/functions/math/det.py [hunk 1]
```diff
@@ -75,8 +72,10 @@ def backward_cpu(self, x, gy):
     def backward_gpu(self, x, gy):
         x, = x
         gy, = gy
-        grad = (gy[:, None, None] * self.detx[:, None, None] *
-                inv._inv_gpu(x.transpose((0, 2, 1))))
+        inv_x, info = inv._inv_gpu(x.transpose((0, 2, 1)))
+        if cuda.cupy.any(info != 0):
+            raise numpy.linalg.LinAlgError('Singular Matrix')
```
Comment (unnonouno) [inline chainer/functions/math/det.py]: Is this error appropriate?

File: chainer/functions/math/det.py [hunk 2]
```diff
@@ -60,9 +59,7 @@ def forward_cpu(self, x):
         return self.detx,
 
     def forward_gpu(self, x):
-        self.detx, success = _det_gpu(x[0])
-        if not success:
```
Comment (unnonouno) [inline chainer/functions/math/det.py]: Can you check if this method throws no error on forward calculation?

File: chainer/functions/math/det.py (hunk 1 above)
Comment (delta2323) [inline chainer/functions/math/det.py]: I changed the raised error to `ValueError`.

File: chainer/functions/math/det.py (hunk 2 above)
Comment (delta2323) [inline chainer/functions/math/det.py]: I added test cases.

File: tests/chainer_tests/functions_tests/math_tests/test_det.py [hunk 3]
```diff
@@ -135,13 +135,28 @@ def test_expect_scalar_gpu(self):
         y = F.det(x)
         self.assertEqual(y.data.ndim, 1)
 
+    def check_singular_matrix(self, x):
+        if self.batched:
```
Comment (unnonouno) [inline tests/chainer_tests/functions_tests/math_tests/test_det.py]: Please add a comment about this test case.

**Comment by unnonouno** [comment]
Please rebase or merge the master.

**Comment by unnonouno** [comment]
LGTM!

**Pull request merged**